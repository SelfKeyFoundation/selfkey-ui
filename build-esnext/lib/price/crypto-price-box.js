import * as React from 'react';
import { withStyles, createStyles } from '@material-ui/core';
import { PriceSummary } from './price-summary';
const styles = createStyles({
    cryptoPriceBox: {
        alignItems: 'flex-start',
        display: 'flex',
        flexDirection: 'column',
        height: 'calc(100% - 36px)',
        justifyContent: 'flex-start',
        marginLeft: '40px',
        marginRight: '8px',
        padding: '16px 0',
    },
    smallText: {
        color: '#fff',
        fontSize: '12px',
        marginTop: '3px !important'
    },
    test: {
        color: 'red'
    }
});
export const CryptoPriceBox = withStyles(styles)(({ classes, locale, cryptoCurrency, cryptoValue, toCurrency, toValue }) => (React.createElement("div", { className: classes.cryptoPriceBox },
    React.createElement(PriceSummary, { className: classes.test, locale: locale, priceStyle: "decimal", currency: cryptoCurrency, value: cryptoValue, appendCurrency: true }),
    React.createElement(PriceSummary, { locale: locale, priceStyle: "currency", currency: toCurrency, value: toValue, appendCurrency: true, valueClass: classes.smallText }))));
export default CryptoPriceBox;
//# sourceMappingURL=crypto-price-box.js.map