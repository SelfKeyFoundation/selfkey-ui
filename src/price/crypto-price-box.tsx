import * as React from 'react';
// @ts-ignore
import injectSheet, { StyleSheet, StyledComponentProps } from 'react-jss';

import { PriceSummary } from './price-summary';

const styles: StyleSheet = {
    cryptoPriceBox: {
        height: 'calc(100% - 36px)',
        padding: '18px 0 18px 0',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'flex-start',
        justifyContent: 'flex-start',
        marginRight: '10px',
        marginLeft: '40px',
    },
    smallText: {
        color: '#fff',
        fontSize: '12px',
        marginTop: '3px !important'
    },
    test: {
        color: 'red'
    }
};

export type CryptoPriceBoxProps = {
    locale: string,
    cryptoCurrency: string,
    cryptoValue: number,
    toCurrency: string
    toValue: number
}

export const CryptoPriceBox = injectSheet(styles)<CryptoPriceBoxProps>(({classes, children, locale, cryptoCurrency, cryptoValue, toCurrency, toValue }) => (
    <div className={classes.cryptoPriceBox}>
        <PriceSummary className={classes.test} locale={locale} priceStyle="decimal" currency={cryptoCurrency} value={cryptoValue} appendCurrency/>
        <PriceSummary locale={locale} priceStyle="currency" currency={toCurrency} value={toValue} appendCurrency valueClass={classes.smallText} />
    </div>
));

export default CryptoPriceBox;
