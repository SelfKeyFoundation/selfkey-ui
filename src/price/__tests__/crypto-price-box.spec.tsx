import * as React from 'react';
import { create } from 'react-test-renderer';
import { CryptoPriceBox } from '../crypto-price-box';

describe('Crypto Price Box', () => {
	it('should show Crypto Price Box for Key', () => {
		const component = create(
			<CryptoPriceBox
				locale="en"
				cryptoCurrency="KEY"
				cryptoValue={4}
				toCurrency="USD"
				toValue={0.0174857}
			/>
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
