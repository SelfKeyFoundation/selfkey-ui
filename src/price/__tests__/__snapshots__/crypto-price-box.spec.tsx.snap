// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crypto Price Box should show Crypto Price Box for Key 1`] = `
<div
  className="Component-cryptoPriceBox-0-1-1"
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 Component-test-0-1-2"
  >
    <div
      className="MuiGrid-item-2 Component-currency-0-1-4"
    >
      KEY
    </div>
    <div
      className="MuiGrid-item-2 Component-value-0-1-5"
    >
      <div>
        4
         
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-align-items-xs-center-9 Component-row-0-1-3"
  >
    <div
      className="MuiGrid-item-2 Component-currency-0-1-4"
    >
      USD
    </div>
    <div
      className="MuiGrid-item-2 Component-value-0-1-5"
    >
      <div>
        $0.0174857
         
      </div>
    </div>
  </div>
</div>
`;
